packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionHierarchy_def.hpp

```diff
@@ -1195,6 +1195,48 @@ MakeCompositeDirectSolver(RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal
   return coarseSolver;
 } // MakeCorseCompositeDirectSolver
 
+/* Create an AMG hierarchy for a composite operator
+ *
+ * Create the hierarchy object and perform the multigrid setup.
+ * Finally, the hierarhcy object will be ready to be applied during the region MG V-cycle call.
+ */
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+RCP<MueLu::Hierarchy<Scalar, LocalOrdinal, GlobalOrdinal, Node> >
+MakeCompositeAMGHierarchy(RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> >& compOp, const std::string& xmlFileName)
+{
+#include "MueLu_UseShortNames.hpp"
+#include "Xpetra_UseShortNames.hpp"
+
+  const Scalar one = Teuchos::ScalarTraits<Scalar>::one();
+
+  RCP<Teuchos::FancyOStream> fos = Teuchos::fancyOStream(Teuchos::rcpFromRef(std::cout));
+  fos->setOutputToRootOnly(0);
+  *fos << "Attempting to setup AMG hierarchy for the composite coarse grid problem" << std::endl;
+
+  // Get parameter list for AMG hierarchy
+  RCP<ParameterList> mueluParams = Teuchos::getParametersFromXmlFile(xmlFileName);
+
+  // Add nullspace information
```

Comment:
We probably also want to add coordinates to do rebalancing on level 0 since we probably inherit a rather small number of unknow per rank at this point.

Reply:
I added a comment to the code, so we won't forget that.